### ISSUE TITLE
Add support for Calendar events (iCalendar vEvent)

### ISSUE THREAD

**Issue by sven-mvolution** [bbottema/simple-java-mail#138]
Hi and thanks a lot for this great library!

My feature request: I would like to use Simple Java Mail to send calendar invitations, which have to smoothly integrate with Gmail/Google calendar and MS Outlook but should also get properly displayed in any ordinary client. This means I have to create multipart/alternative messages consisting of 3! alternatives: text/plain, text/html and text/calendar - at least MS Outlook is sending such invitations. Simply adding the ICS as attachment does not work with Outlook.

I found no way how to do this with Simple Java Mail, e.g. there is no `setTextVCalendar()` or `addCustomText(String text, String mimeType)`.

What do you think?

**Comment by bbottema** [comment]
Hi @sven-mvolution, that's a good point! I will think about a proper solution.

**Comment by bbottema** [comment]
I've been trying to create a MimeMessage by hand with an ical event in it using [ical4j](https://github.com/ical4j/ical4j), but I've been unable to get gMail or Outlook to cooperate with me. It keeps seeing it as a simple attachment. I've not been able to test it in Outlook yet (don't have Outlook).

I'll keep playing around.

**Comment by sven-mvolution** [comment]
Hi @bbottema - thanks for further investigating and considering this a useful enhancement!
Meanwhile I've locally changed your lib to suit my needs as follows - but of course still waiting for the "official" support ;)
- Add support for a optional text/calendar alternative
- Use 2051 test port and made unit test work
See the attached patch file (diff against 5.0.3)
[calendar-alternative.patch.txt](https://github.com/bbottema/simple-java-mail/files/2063553/calendar-alternative.patch.txt)

**Comment by bbottema** [comment]
Some how Simple Java Mail manages the bodyparts better than when I do it manually :D

It works in Outlook 365 for me, but gMail still refused and only displays it as an attachment. I wonder if it even still works at all in gMail.

**Comment by bbottema** [comment]
Implemented for the 6.0.0 release.

Usage is simple:

```
// Create a Calendar with ical4j
Calendar icsCalendar = new Calendar();
icsCalendar.getProperties().add(new ProdId("-//Events Calendar//iCal4j 1.0//EN"));
icsCalendar.getProperties().add(Version.VERSION_2_0);

(..) // add attendees, organizer, end/start date and whatever else you need

// Produce calendar string
ByteArrayOutputStream bOutStream = new ByteArrayOutputStream();
new CalendarOutputter().output(icsCalendar, bOutStream);
String yourICalEventString = bOutStream.toString("UTF-8")

// Let Simple Java Mail handle the rest
currentEmailBuilder
    .withCalendarText(CalendarMethod.REQUEST, yourICalEventString)
```

**Comment by bbottema** [comment]
Released in 6.0.0-rc1.

**Comment by bbottema** [comment]
6.0.0 has released as well, finally.
